docs/api/isAnyOf.mdx

```diff
@@ -0,0 +1,111 @@
+---
+id: isAnyOf
+title: isAnyOf
+sidebar_label: isAnyOf
+hide_title: true
+---
+
+# `isAnyOf`
+
+A higher-order function that accepts several type guard functions or `redux-toolkit` action creator functions and
+returns a type guard function that matches (returns true) if *any* of the provided functions match.  Action creator
+functions include the ones produced by `createAction` or the child actions of an async thunk produced by
+`createAsyncThunk`, such as `asyncThunk.pending`, `asyncThunk.fulfilled`, and `asyncThunk.rejected`.
+
+```js
+function isAnyOf(...matchers)
+```
+
+## Reducing code duplication and boilerplate
+
+When using the `builder` pattern to construct a reducer, we add cases or matchers one at a time.  However,
+sometimes we can use the same reducer for several cases.  The following is a trivial example.
+
+```ts
+import { createAsyncThunk, createReducer } from '@reduxjs/toolkit'
+
+interface LoadingState {
+  isLoading: boolean;
+}
+
+const initialState: LoadingState = { isLoading: false }
+
+const request = createAsyncThunk('request', () => {});
+
+const loadingReducer = createReducer(initialState, builder => {
+  builder
+    .addCase(request.pending, (state, action) => ({
+      isLoading: true
+    }))
+    .addCase(request.fulfilled, (state, action) => ({
+      isLoading: false
+    }))
+    .addCase(request.rejected, (state, action) => ({
+      isLoading: false
+    }))
+})
```

Comment:
I would consider keeping your `return` uniform throughout the examples - some folks are definitely going to get tripped up between implicit and explicit returns as well as mutable vs immutable updates. Being that we push `immer` so much, I'd recommend we just show:

```js
(state, action) => {
   state.isLoading = false;
}
```

Reply:
Good idea.  Thanks!